tests/unit/test_sagemaker_steps.py

```diff
@@ -265,6 +267,78 @@ def test_training_step_creation(pca_estimator):
     }
 
 
+@patch('botocore.client.BaseClient._make_api_call', new=mock_boto_api_call)
+@patch.object(boto3.session.Session, 'region_name', 'us-east-1')
+def test_training_step_creation_with_placeholders(pca_estimator):
+    execution_input = ExecutionInput(schema={
+        'Data': str,
+        'JobName': str,
+        'OutputPath': str,
+    })
```

Comment:
ExecutionInput is just one type of placeholder. Add StepInput placeholders to test that that works too.

Reply:
Good point! Will add a test case for StepInput in the next commit!

Reply:
You can combine them in this test case:

```python
    execution_input = ExecutionInput(schema={
        'Data': str,
        'OutputPath': str,
    })

    step_input = ExecutionInput(schema={
        'JobName': str,
    })

   # ...
   job_name=step_input['JobName']

   #... assert
   'TrainingJobName.$': ...
```

Reply:
Good idea! Will do in the next commit